### PR TITLE
Add KVCache trans for long sequence && tuned comm for faster Addreduce

### PR DESCRIPTION
**Pull request by abenmao**
Transpose KVCache with env var "ENABLE_KV_TRANS" for long sequence
Tune addreduce between shm and ccl for faster comm

| workload | baseline-2s | opt-2s |
| --------------- | --------------- | --------------- |
| llama-2-13b bs24 in512 bf16 (SPR+HBM) | 350ms | 90ms |
| llama-2-70b bs32 in4096 bf16 (SPR) | 1300ms | 640ms |

### CODE REVIEW

File: src/common/kvcache_tensor.h [hunk 1]
```diff
@@ -21,6 +21,8 @@
 
 #include "allocator.h"
 
+extern bool kvTrans();
+
 /**
  * Tensor specially designed for KV Cache
  * Naturaly, it could be represented in the shape of [seq_length][batch_size][head_num][head_size]
```
Comment (pujiang2018) [inline src/common/kvcache_tensor.h:28]: Please also modify the comments here?

File: src/models/env_config.cpp [hunk 2]
```diff
@@ -18,14 +18,41 @@
...
+        if (skipMsk == 1) printf("ENABLE_SKIP_MASK is enabled for ignoring mask Q*K.\n");
+    }
+    return skipMsk == 1;
+}
+
+bool kvTrans() {
+    static int kvTrans = -1;
+    if (kvTrans == -1) {
+        kvTrans = (getenv("ENABLE_KV_TRANS") ? atoi(getenv("ENABLE_KV_TRANS")) : 0);
+        // if (kvTrans == 1)
```
Comment (pujiang2018) [inline src/models/env_config.cpp:54]: please remove it if not used.

File: src/models/env_config.cpp [hunk 3]
```diff
@@ -18,14 +18,41 @@
 
 bool enableCATMLP() {
     static int catMlp = -1;
-    if (catMlp == -1)
-        catMlp = (getenv("ENABLE_CAT_MLP") ? atoi(getenv("ENABLE_CAT_MLP")) : 1);
+    if (catMlp == -1) catMlp = (getenv("ENABLE_CAT_MLP") ? atoi(getenv("ENABLE_CAT_MLP")) : 1);
     return catMlp == 1;
 }
 
+bool tunedComm() {
```
Comment (pujiang2018) [inline src/models/env_config.cpp:25]: it is not so easy to understand "Tuned communication". add some comment?

Comment (abenmao) [reply src/models/env_config.cpp:26]: Added~

File: src/models/env_config.cpp (hunk 2 above)
Comment (abenmao) [reply src/models/env_config.cpp]: Added some comments.

File: src/common/kvcache_tensor.h (hunk 1 above)
Comment (abenmao) [reply src/common/kvcache_tensor.h:28]: Done~ New layout default disabled.

**Review by pujiang2018 — APPROVED** [review]

**Pull request merged**